app/components/graph-editor/drawable-element.ts

```diff
@@ -0,0 +1,275 @@
+// File: drawable-element.ts
+// Created by: CJ Dimaano
+// Date created: February 4, 2017
+
+
+import { FONT_SIZE } from "./defaults";
+import { GraphEditorCanvas, point, rect, size } from "./graph-editor-canvas";
+import { Drawable } from "./drawable";
+import { DrawableGraph } from "./drawable-graph";
+import {
+    PropertyChangedEventArgs,
+    PropertyChangedEventEmitter,
+    PropertyChangedEventListener
+} from "./events";
+
+
+/**
+ * DrawableStates  
+ *   The states in which a drawable can be.
+ */
+export enum DrawableStates {
+    Default,
+    Dragging,
+    Hovered
+};
+
+/**
+ * DrawableElement  
+ *   Common properties shared between drawable edges and nodes.
+ */
+export abstract class DrawableElement extends Drawable {
```

Comment:
This seems pretty complicated. Would it be possible to avoid an abstract class by pulling this out into an interface and a putting the logic into a common class/function?